### ISSUE TITLE
Connect Project with the git repository hosting

### ISSUE THREAD

**Issue by ZakariaTalhami** [saurabhdaware/projectman#14]
It would be useful to associate a project with like a github repository, then users would be able to open the repository page in the browser using projectman directly.

**Suggestions:**
`pm addrepo [projectName] [--url=<url>]`
This would prompt the user if arguements are not specified, and add the url into the *settings.json* file with the project meta.

**Open the repository page:**
`pm openRepo [projectName] [--browser=<browser>]`
Opens with default browser or the browser given if specified.

This added infromation might be useful in the future, for example, if you wanted to inplement a clone feature, or what not.

**Comment by saurabhdaware** [comment]
That sounds like a great idea! 
We can have `addrepo` command but instead of having `openrepo` we can put them inside normal projects with a `(Repo)` appended to the name when showing in open so we won't have to separately document another command. 

Also, I don't think so there's any need of having a `--browser` right now opening it in default browser should work for all.

**Comment by ZakariaTalhami** [comment]
> I don't think so there's any need of having a `--browser` right now

Yeah this would work just fine.



> We can have `addrepo` command but instead of having `openrepo` we can put them inside normal projects with a `(Repo)` appended to the name when showing in open so we won't have to separately document another command.

This i might get confusing when opening a project, as there would be urls also.

**Comment by saurabhdaware** [comment]
> This i might get confusing when opening a project, as there would be urls also.

How about having `(URL)` then?

**Comment by ZakariaTalhami** [comment]
i think the simplest way is to add an attribute to the already existing project
```
{
    "name": "projectname" 
    "path": "projectPath" 
    "repo": "projectReo" 
}
```

so `addrepo` would add the field `repo` to the existing project, and use this went opening, but this would require a new command, or a new option to the `open` command.

what do you think?

**Comment by saurabhdaware** [comment]
In that case, people would have to remember and type a new command 

and the `pm open` command already has alias `pm o` and `pm` and these commands are already known and some might be used to typing `pm open`, so it would be annoying for them to `pm open` and then see projects then exit and type `pm openrepo`.

Would be more comfortable for them if they can just remember one command and do everything from the same.

---

Even about `pm addrepo`, I think we can have `pm add --repo` instead and while storing it we can just store it normally inside the path
```js
"projects": [
	{
	    "path": "/home/saurabh/Desktop/projects/project1",
	    "name": "Project1"
	},
	{
	    "path": "https://github.com/saurabhdaware/project2",
	    "name": "Project2"
	}
]
```
So if someone prefers to `pm edit` and change the settings, he can simply put website link in path so he won't have to read docs to see what the key is.

and when user `pm open` he will see:
```
Select Projects to Open:
> Project1
  Project2 (github)
```
so it will show `(<websiteName>)` beside the project.

----

This will make it easier for me to explain the `projectman` to new people and even the existing users will not have to remember any extra command.

**Comment by ZakariaTalhami** [comment]
How would `pm open` recognize when a project is a actual project and open in editor and when to open in browser?

**Comment by saurabhdaware** [comment]
with regex we can check if the path is link or the local path. If it has https or http then we will open it in browser

**Comment by ZakariaTalhami** [comment]
I would be willing to have a go at this.

**Comment by saurabhdaware** [comment]
That sounds great! go ahead :D

**Comment by saurabhdaware** [comment]
Resolved in #20  
Thanks 🎉
